## Airport and Weather Data Analysis - Assignment 5
This notebook analyzes the northernmost and easternmost airports in the United States, and identifies which New York area airport had the windiest weather on February 12, 2013. The data used comes from the `nycflights13` dataset, specifically the `airports.csv` and `weather.csv` files.


In [6]:
import pandas as pd

# Load the datasets
airports_df = pd.read_csv('C:/Users/since/Downloads/airports.csv')
weather_df = pd.read_csv('C:/Users/since/Downloads/weather.csv')



In [12]:
# Northernmost Airport
northernmost_airport = airports_df.loc[airports_df['lat'].idxmax()]
print("Northernmost Airport:", northernmost_airport['name'], "FAA Code:", northernmost_airport['faa'])

# Easternmost Airport
easternmost_airport = airports_df.loc[airports_df['lon'].idxmax()]
print("Easternmost Airport:", easternmost_airport['name'], "FAA Code:", easternmost_airport['faa'])


Northernmost Airport: Dillant Hopkins Airport FAA Code: EEN
Easternmost Airport: Eareckson As FAA Code: SYA


### Results:
- The **northernmost airport** is **Dillant Hopkins Airport** (FAA Code: EEN), located at latitude 72.27.
- The **easternmost airport** is **Eareckson As** (FAA Code: SYA), located at longitude 174.11.


In [15]:
# Filter weather data for NY area airports on February 12, 2013
ny_airports = ['JFK', 'LGA', 'EWR']
feb_12_weather = weather_df[
    (weather_df['origin'].isin(ny_airports)) &
    (weather_df['year'] == 2013) &
    (weather_df['month'] == 2) &
    (weather_df['day'] == 12)
]

# Find the maximum wind speed per airport
max_wind_per_airport = feb_12_weather.groupby('origin')['wind_speed'].max()
print(max_wind_per_airport)


origin
EWR    1048.36058
JFK      20.71404
LGA      23.01560
Name: wind_speed, dtype: float64


### Windiest Airport on February 12, 2013:
- The **windiest airport** in the New York area was **LaGuardia (LGA)** with a wind speed of 23.02 m/s.


## Assumptions and Adjustments
- The wind speed for Newark (EWR) had an unusually high value, which was treated as an outlier and disregarded for the purposes of this analysis.

- The wind speed for Newark (EWR) was an unusually high value (1048.36 m/s), which is unrealistic and was therefore treated as a data entry error and disregarded.



## Conclusion
This analysis provided insights into U.S. airport geography and weather patterns. The northernmost airport is Dillant Hopkins Airport (EEN), and the easternmost is Eareckson As (SYA). Additionally, LaGuardia (LGA) had the highest wind speed of 23.02 m/s on February 12, 2013.
